test/validation/api/random/random.c

```diff
@@ -9,11 +9,25 @@
 
 static void random_test_get_size(void)
 {
-	int32_t ret;
+	/* odp_random_data may fail to return data on every call (i.e. lack of
+	 * entropy). Therefore loop with some sane loop timeout value. Note that
+	 * it is not required for implementation to return data in the "timeout"
+	 * amount of steps. Rather it is a way for preventing the test to loop
+	 * forever.
+	 */
+	int32_t ret, timeout = 100;
+	uint32_t bytes = 0;
 	uint8_t buf[32];
 
-	ret = odp_random_data(buf, sizeof(buf), ODP_RANDOM_BASIC);
-	CU_ASSERT(ret == sizeof(buf));
+	do {
+		ret = odp_random_data(buf, sizeof(buf), ODP_RANDOM_BASIC);
+		if (ret < 0)
+			break;
+		bytes += ret;
+	} while (bytes < sizeof(buf) && timeout > 0);
```

Comment:
Two points:

1. Since `timeout` is never changed from its initial value of 100, how is the `timeout > 0` clause adding anything to this test?

2. This accumulates the bytes returned, but if less than the initial request is returned, subsequent calls overwrite `buf`, meaning it's not properly filled. Need to increment the point where new bytes are inserted into `buf` as well to successfully accumulate the requested number of bytes.

Reply:
Correct. Will fix in V2.